mesonbuild/dependencies/pkgconfig.py

```diff
@@ -89,13 +89,13 @@ def __init__(self, name: str, environment: 'Environment', kwargs: T.Dict[str, T.
         assert isinstance(self.pkgbin, ExternalProgram)
         mlog.debug('Determining dependency {!r} with pkg-config executable '
                    '{!r}'.format(name, self.pkgbin.get_path()))
-        ret, self.version, _ = self._call_pkgbin(['--modversion', name])
-        if ret != 0:
-            return
+        try:
+            ret, self.version, err = self._call_pkgbin(['--modversion', name])
+            if ret != 0:
+                raise DependencyException(f'Could not query version for {name}:\n{err}\n')
```

Comment:
The logfile will now log this as:

```
f"pkg-config error with '{name}': Could not query version for {name}:\n{err}\n"
```

Which seems a bit awkward...

I also think we probably don't care about specifically mentioning that we failed to query the version, it's just a general error trying to find it with pkg-config at all...

Reply:
In `_call_pkgbin_real`, we do this:

```
mlog.debug(f"Called `{call}` -> {rc}\n{out}")
return rc, out, err
```

If we just logged err there as well, then the "relevant bits from meson-log.txt" would indeed tell you what the problem was, albeit not emit it directly to the console. Is that an acceptable solution?

Reply:
I am not familiar with meson codebase and I wasn't thinking of that when implementing the change, though I would much prefer if I could see the transitive dependency name on console, as looking at meson-log.txt can be much more work. For example I do call `meson setup` as part of a docker build and when that fails the incomplete docker image is thrown away and I would need to run the the image from previous stage and run the exact same meson command and only then I can look into meson-log.txt, which is even more steps than calling the pkg-config command from inside that image.

Reply:
There may be context in that log above and beyond pkg-config calls. What I usually do in things like CI logs is something like:
```python
if setup_failed:
    log_file = Path(builddir, 'meson-logs', 'meson-log.txt')
    print('::group::==== meson-log.txt ====')
    print(log_file.read_text(encoding='utf-8'))
    print('::endgroup::')
```

(The group markers are there for Github to make it fold until you click on it to expand it.)

Reply:
Sure everything can be solved with a layer of indirection, I do prefer to call meson directly though rather than via a wrapper script. Also in this case, transitive dependency name is the most important part of the error message, as I need to take action like installing it or updating PKG_CONFIG_PATH with it.